site/app/controllers/grading/ElectronicGraderController.php

```diff
@@ -47,7 +47,53 @@ private function validateDiffViewerType(string $type) {
         }
         return true;
     }
-
+/**
+        * Route for randomizing peer assignments
+        * @Route("/{_semester}/{_course}/gradeable/{gradeable_id}/ajaxRandomizePeers", methods={"POST"})
+        * @AccessControl(role="INSTRUCTOR")
+    */ 
+    public function ajaxRandomizePeers($gradeable_id){
+        $number_to_grade=1;
+       // $gradeable_id=$_POST['gradeable_id'];
+        $gradeable = $this->tryGetGradeable($gradeable_id);
+         if ($gradeable === false) {
+             $this->core->addErrorMessage('Invalid Gradeable!');
+             $this->core->redirect($this->core->buildCourseUrl());
+         }
+         $order = new GradingOrder($this->core, $gradeable, $this->core->getUser(), true);
+         $student_array= [];
+         $student_list= [];
+         $students = $this->core->getQueries()->getUsersByRegistrationSections($order->getSectionNames());
+         foreach ($students as $student) {
+              $reg_sec = ($student->getRegistrationSection() === null) ? 'NULL' : $student->getRegistrationSection();
+              $sorted_students[$reg_sec][] = $student;
+              array_push($student_list, [
+                  'user_id' => $student->getId(),
+                 // reg_section => $reg_sec,
+              ]);
+              array_push($student_array,$student->getId());
+          }
+          $graded_array=$student_array;
+          $final_grading_info=[];
+          foreach($student_array as $grader){
+              $temp_array=$graded_array;
+              if (($key = array_search($grader, $temp_array)) !== false) {
+                unset($temp_array[$key]);
+            }
+              shuffle($temp_array);
+              $grading_list=[];
+              for($i=0;$i<$number_to_grade;++$i){
+                 array_push($grading_list,$temp_array[$i]);
+                 if (($key = array_search($temp_array[$i], $graded_array)) !== false) {
+                    unset($graded_array[$key]);
+                }
+              }
+              array_push($final_grading_info,[$grader,$grading_list]);
+          }
+         $gradeable->setRandomPeerGradersList($final_grading_info);
```

Comment:
I don't understand why we need a special`setRandomPeerGradersList`. We should just have a `setPeerGradersList` which takes in a list of assignments and sets it. All special logic necessary to generate that list should occur elsewhere.